Adding custom scalars

marshalWilliam opened the pull request:
Added custom scalars (Binary, EmailAddress, DateTime)

marshalWilliam committed:
Adding custom scalars

Added custom scalars (binary, emailaddress, datetime)

ccpacillos commented on schemas/scalars/datetime.ts:
```diff
@@ -0,0 +1,21 @@
+        `Query Error: Can only parse strings as datetime but got a: ${ast.kind}`
+      );
+    }
+
```
Add validation here to check if input string is a valid ISO date string.

marshalWilliam replied:
Good afternoon po sir, I've added the validation sir. Thank you sir!

marshalWilliam committed:
Added ISO date string validation

Added a validation in datetime custom scalar to check is input is a valid ISO date string

marshalWilliam committed:
Added Mutation.signUp

ccpacillos commented on schemas/scalars/datetime.ts:
```diff
@@ -0,0 +1,28 @@
+      );
+    }
+
+    if (ast.value !== new Date(Date.parse(ast.value)).toISOString()) {
```
There's a better way of checking this. You can refer to similar solutions here: https://thewebdev.info/2021/03/27/how-to-check-if-a-string-is-a-date-string-with-javascript

marshalWilliam replied:
Noted po sir. How about this one po.

marshalWilliam committed:
Changed datetime validation
